### PR TITLE
Fix speed when move in diagonal direction and fix the sound when player take damage

### PR DESCRIPTION
**Pull request by VuKhanhVyHo**
# Description

Sounds are played when the player takes damage from the NPC. Before the player moving faster in diagonal direction, which is fixed.

Fixes / Closes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Fix speed when players moving diagonally
- [ ] Sounds added when players take damage

# How Has This Been Tested?
Testing manually using UI

# Checklist:

- [ ] The players now have sounds when take damage from NPC
- [ ] I have performed a self-review of my code
- [ ] The players moving at a fixed speed even though in the diagonal direction

### CODE REVIEW

**Review by Jordamnmn — APPROVED** [review]
Looks good but it's inevitably going to cause conflicts as other member merge their code into main

**Review by mblauberg — APPROVED** [review]
The code looks solid overall. I noticed that this PR also includes a lot of refactors and local variable name changes in unrelated files. So it may be a good idea to update the description to mention this to give full context to future reviewers. Nice work though!

File: source/core/src/main/com/csse3200/game/components/player/KeyboardPlayerInputComponent.java [hunk 1]
```diff
@@ -83,17 +87,57 @@ public KeyboardPlayerInputComponent() {
...
+            walkDirection.x += 1;
+        }
+
+        if (walkDirection.len() > 0) {
+            walkDirection.nor();  // Normalize to prevent faster diagonal movement
+            entity.getEvents().trigger("walk", walkDirection);
+        } else {
+            entity.getEvents().trigger("walkStop");
+        }
+    }
```
Comment (manya-k) [inline source/core/src/main/com/csse3200/game/components/player/KeyboardPlayerInputComponent.java:138]: none of the player animations are working in this branch

**Review by manya-k — CHANGES_REQUESTED** [review]
Functionality tested. 
none of the player animations are working . please fix them before the merge.

this branch has merged ui-refactor into it which had to be reverted from main so this branch cannot go in before ui-refactor.

**Comment by manya-k** [comment]
please fix the bugs and create a new pr when resolved so that other branches can be merged

File: source/core/src/main/com/csse3200/game/components/player/KeyboardPlayerInputComponent.java (hunk 1 above)
Comment (manya-k) [reply source/core/src/main/com/csse3200/game/components/player/KeyboardPlayerInputComponent.java:138]: when u go left or right, the left walk or right walk animations dont work

**Pull request closed without merging**